modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/util/fieldSupport.es.js

```diff
@@ -23,6 +23,97 @@ import {
 import {updateField} from '../components/LayoutProvider/util/settingsContext.es';
 import {FIELD_TYPE_FIELDSET} from './constants.es';
 
+export const addField = ({
+	defaultLanguageId,
+	editingLanguageId,
+	fieldNameGenerator,
+	generateFieldNameUsingFieldLabel,
+	indexes,
+	newField,
+	pages,
+	parentFieldName,
+}) => {
+	const {columnIndex, pageIndex, rowIndex} = indexes;
+
+	let newPages;
+
+	if (parentFieldName) {
+		const visitor = new PagesVisitor(pages);
+
+		newPages = visitor.mapFields(
+			(field) => {
+				if (field.fieldName === parentFieldName) {
+					const nestedFields = field.nestedFields
+						? [...field.nestedFields, newField]
+						: [newField];
+
+					field = updateField(
+						{
+							defaultLanguageId,
+							editingLanguageId,
+							fieldNameGenerator,
+							generateFieldNameUsingFieldLabel,
+						},
+						field,
+						'nestedFields',
+						nestedFields
+					);
+
+					const {rows} = field;
+					const pages = FormSupport.addFieldToColumn(
+						[
+							{
+								rows:
+									typeof rows === 'string'
+										? JSON.parse(rows)
+										: rows,
```

Comment:
If I'm not mistaken, it shouldn't have come as a string. Probably the backend is bringing this serialized. Maybe the Data Engine brings this as a string and Forms does not.